base/uk.ac.stfc.isis.ibex.opis/check_opi_format.py

```diff
@@ -0,0 +1,272 @@
+                               "/widget[@typeId='org.csstudio.opibuilder.widgets.Label']/text"
+
+        for name in root.xpath(container_name_xpath):
+
+            words = name.text.split()
+
+            for word in words:
+                capitalisation_error = False
+                original_word = word
+
+                if word is words[0]:
+                    cased_word = word.title()
+                else:
+                    cased_word = word.lower()
+
+                if (original_word != cased_word) \
+                        and not any(s in original_word for s in self.ignore) \
+                        and len(original_word) > self.ignore_short_words_limit:
+                    capitalisation_error = True
+
+                if capitalisation_error:
+                    err = "Error on line " + str(name.sourceline) + ": " + etree.tostring(name) \
+                          + "\n... Labels should be in 'sentence case' " \
+                            "(Expected '" + cased_word + "', but got '" + original_word + "')."
+                    self.errors.append(err)
+
+    def check_colon_at_the_end_of_labels(self, root):
+        container_name_xpath = "//widget[@typeId='org.csstudio.opibuilder.widgets.groupingContainer']" \
+                               "/widget[@typeId='org.csstudio.opibuilder.widgets.Label']/text"
+
+        for name in root.xpath(container_name_xpath):
+
+            words = name.text
+
+            # Check last character in the string is a colon
+            if words[-1:] != ":" and words[-3:] != "..." and not words.isdigit() \
+                    and not any(s in words for s in self.ignore):
+                err = "Error on line " + str(name.sourceline) + ": " + etree.tostring(name) \
+                      + "\n... Labels should usually have a colon at the end, unless this is a tabular layout"
+                self.errors.append(err)
+
+    def run(self):
+
+        self.errors = []
+        build_contains_errors = False
+
+        for filepath in self.file_iterator():
+
+            root = self.get_tree(filepath)
+
+            self.check_led_colours(root)
+            self.check_opi_label_fonts(root)
+            self.check_plot_area_background(root)
+            self.check_text_input_colors(root)
+            self.check_items_are_in_grouping_containers(root)
+            self.check_capitals_for_grouping_containers(root)
+            self.check_capitals_for_labels(root)
+            self.check_capitals_for_labels_outside_grouping_containers(root)
+            self.check_colon_at_the_end_of_labels(root)
+
+            if len(self.errors) > 0:
+                print "\n --------------"
+                print str(len(self.errors)) + " errors found in file " + filepath
+                print " --------------\n"
+                build_contains_errors = True
+                for error in self.errors:
+                    print error
```

Comment:
Write all the results to a txt file rather than (or in addition to) printing it to the console. Also I think it would be nice to have a little summary at the end, like x out of y files had errors.